azurerm/internal/services/storage/storage_data_lake_gen2_filesystem_resource.go

```diff
@@ -292,24 +296,19 @@ func resourceStorageDataLakeGen2FileSystemRead(d *schema.ResourceData, meta inte
 		return fmt.Errorf("Error setting `properties`: %+v", err)
 	}
 
-	// The above `getStatus` API request doesn't return the ACLs
-	// Have to make a `getAccessControl` request, but that doesn't return all fields either!
-	pathResponse, err := pathClient.GetProperties(ctx, id.AccountName, id.DirectoryName, "/", paths.GetPropertiesActionGetAccessControl)
-	if err != nil {
-		if utils.ResponseWasNotFound(pathResponse.Response) {
-			log.Printf("[INFO] Root path does not exist in File System %q in Storage Account %q - removing from state...", id.DirectoryName, id.AccountName)
-			d.SetId("")
-			return nil
+	// acl is only enabled when `IsHnsEnabled` is true otherwise the rest api will report error
+	if storageAccount.IsHnsEnabled != nil && *storageAccount.IsHnsEnabled {
+		// The above `getStatus` API request doesn't return the ACLs
+		// Have to make a `getAccessControl` request, but that doesn't return all fields either!
+		pathResponse, err := pathClient.GetProperties(ctx, id.AccountName, id.DirectoryName, "/", paths.GetPropertiesActionGetAccessControl)
+		if err == nil {
+			acl, err := accesscontrol.ParseACL(pathResponse.ACL)
+			if err != nil {
+				return fmt.Errorf("Error parsing response ACL %q: %s", pathResponse.ACL, err)
+			}
+			d.Set("ace", FlattenDataLakeGen2AceList(acl))
```

Comment:
"ace" should always be set to a value, so we should assign `FlattenDataLakeGen2AceList` to an `[]interface{}` variable and then always set `ace` so that we can display drift here